stylix/testbed.nix

```diff
@@ -0,0 +1,95 @@
+{ pkgs, inputs, lib, ... }:
+
+let
+  commonModule = { config, ... }: {
+    users.users.testbed = {
+      description = "Testbed";
+      hashedPassword = "";
+      isNormalUser = true;
+    };
+
+    # The state version can safely track the latest release because the disk
+    # image is ephermal.
+    system.stateVersion = config.system.nixos.release;
+    home-manager.users.testbed.home.stateVersion = config.system.nixos.release;
+
+    virtualisation.vmVariant.virtualisation = {
+      # This is a maximum limit; the VM should still work if the host has fewer cores.
+      cores = 4;
+      memorySize = 2048;
+    };
+  };
+
+  autoload = builtins.concatLists
+    (lib.mapAttrsToList
+      (name: _:
+        let testbed = {
+          inherit name;
+          module = "${../modules}/${name}/testbed.nix";
+        };
+        in
+          lib.optional (builtins.pathExists testbed.module) testbed
+      )
+      (builtins.readDir ../modules));
```

Comment:
This function mimics what we do to load real modules, too - see `stylix/autoload.nix`.

The difference in this version is that it also returns the folder name so that we can name the package output accordingly.